<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="jaeger">
  <!-- Link “dummy” para satisfacer robot_state_publisher -->
  <link name="base_link"/>
  <link name="x_link">
    <inertial>
      <mass value="0.001"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6"/>
    </inertial>
  </link>
  
  <joint name="x_joint" type="continuous">
    <parent link="base_link"/>
    <child link="x_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <link name="y_link">
    <inertial>
      <mass value="0.001"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="y_joint" type="continuous">
    <parent link="base_link"/>
    <child link="y_link"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
  </joint>

  <!-- Tu ros2_control -->
  <ros2_control name="MyRobotHardware" type="system">
    <hardware>.
      <plugin>jaeger_model/MyRobotHardware</plugin>
      <parameter name="update_rate" value="50" />
    </hardware>

    <joint name="x_joint">
        <state_interface name="position"> </state_interface>
    </joint>
    <joint name="y_joint">
        <state_interface name="position"> </state_interface>
    </joint>

    <gpio name="valve1">
        <command_interface name="digital"></command_interface>
    </gpio>
    <gpio name="valve2">
        <command_interface name="digital"></command_interface>
    </gpio>
    <gpio name="valve3">
        <command_interface name="digital"></command_interface>
    </gpio>
    <gpio name="valve4">
        <command_interface name="digital"></command_interface>
    </gpio>

  </ros2_control>
</robot>
